fix(o3r): replaced path.resolve with path.join for better path resolution

gavadhany opened the pull request:
## Proposed change
Change path resolver from path.resolve to path.join  as the tree.exists is unable to resolve the absolute path generated by path.resolve

## Related issues

- :bug: Fixes #1778

matthieu-crouzet commented on packages/@o3r/core/schematics/service/index.ts:
```diff
@@ -46,10 +46,10 @@ function ngGenerateServiceFn(options: NgGenerateServiceSchematicsSchema): Rule {
     if (workspaceProject?.projectType !== 'application') {
       let rec = '..';
-      while (!tree.exists(path.resolve(destination, rec, 'package.json')) && path.resolve(destination, rec) !== '/') {
+      while (!tree.exists(path.join(destination, rec, 'package.json')) && path.resolve(destination, rec) !== '/') {
```
```suggestion
      while (!tree.exists(path.join(destination, rec, 'package.json')) && path.join(destination, rec) !== '/') {
```

gavadhany replied:
@matthieu-crouzet can't use  "path.join" while comparing with root path as we need absolute path. Path.join returns relative path. Hence using path.resolve

matthieu-crouzet replied:
But you are comparing with the tree path that is a relative path to the project 
on Windows you will never have a path matching `'/'`

gavadhany replied:
Thats correct.. so made changes to accept both "Drive letter" and "/" in todays [commit](https://github.com/AmadeusITGroup/otter/compare/f97502a894f140f7f559a34a173703c218b6bac5..771e53a3031a92b7d84421ec93e74a7123d613c8)

gavadhany replied:
Cool.. will take a look prevent it from resolving paths outside projects

gavadhany commented:
@matthieu-crouzet and @kpanot : I have refactored the code a bit. The current version of the code wouldnt work if package.json was in the path entered by user e.g. './'. 
Also added a check to prevent path resolution outside the project workspace and an error package.json doesnt exist within the project workspace.

kpanot commented on packages/@o3r/core/schematics/service/index.ts:
```diff
@@ -45,11 +44,16 @@ function ngGenerateServiceFn(options: NgGenerateServiceSchematicsSchema): Rule {
-      while (!tree.exists(path.resolve(destination, rec, 'package.json')) && path.resolve(destination, rec) !== '/') {
-        rec = path.join('..', rec);
+      let currentDirectory = path.join(destination);
+      let parent = '..';
```
the value never change, it should be `const`.
(I am surprised it has not been caught by the linter :S)

gavadhany replied:
Fixed

matthieu-crouzet commented on packages/@o3r/core/schematics/service/index.ts:
```diff
@@ -45,11 +44,16 @@ function ngGenerateServiceFn(options: NgGenerateServiceSchematicsSchema): Rule {
+        currentDirectory = path.join(currentDirectory, parent);
+      }
+      if (currentDirectory === parent) {
+        throw new SchematicsException('Could not find package.json in the project directory.');
```
```suggestion
        throw new O3rCliError('Could not find package.json in the project directory.');
```
from `@o3r/schematics`

gavadhany replied:
done

kpanot commented on packages/@o3r/core/schematics/service/index.ts:
```diff
@@ -45,11 +44,16 @@ function ngGenerateServiceFn(options: NgGenerateServiceSchematicsSchema): Rule {
+      let currentDirectory = path.join(destination);
+      let parent = '..';
+      // Find the package.json file in the project directory
+      while (currentDirectory !== parent && !tree.exists(path.join(currentDirectory, 'package.json'))) {
```
did you test that `currentDirectory !== parent` is correctly executing in the different OS?
It seems to be quite weak condition, maybe a `.include(parent)` would be a bit stronger?

gavadhany replied:
@kpanot : Changed it from === to includes

gavadhany commented on packages/@o3r/core/schematics/service/index.ts:
```diff
@@ -45,11 +44,16 @@ function ngGenerateServiceFn(options: NgGenerateServiceSchematicsSchema): Rule {
-      let rec = '..';
-      while (!tree.exists(path.resolve(destination, rec, 'package.json')) && path.resolve(destination, rec) !== '/') {
-        rec = path.join('..', rec);
+      let currentDirectory = path.join(destination);
```
Yea.. wanted to normalize. Replaced it with normalize
